Build failure for Base >= 4.11

I tried to install hatex-guide with Stack (lts-12.9) but the build failed with the following error:

```
Text/LaTeX/Guide/Syntax.hs:77:10: error:
• No instance for (GHC.Base.Semigroup Syntax) arising from the superclasses of an instance declaration
• In the instance declaration for ‘Monoid Syntax’
     |
77 | instance Monoid Syntax where
     |          ^^^^^^^^^^^^^
```


This seems to happen because of the change in version 4.11 of base, in which Semigroup is now a superclass of Monoid. Two possible fixes come to mind:

- Write a Semigroup instance for Syntax
- Restrict the version of base in the project .cabal file. Currently, the restriction is too lax: base = 4.*

--- a/Text/LaTeX/Guide/Syntax.hs
+++ b/Text/LaTeX/Guide/Syntax.hs
@@ -1,5 +1,6 @@
 
 {-# LANGUAGE NoImplicitPrelude, OverloadedStrings #-}
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fno-warn-unused-do-bind #-}
 
 module Text.LaTeX.Guide.Syntax (
@@ -17,7 +18,12 @@ import Data.Text
 import Data.Text.IO hiding (putStr)
 import Data.Int
 import Data.Bool
-import Prelude (Eq(..),Show(..),FilePath,Enum,Semigroup)
+import Prelude (Eq(..),Show(..),FilePath,Enum)
+
+#if MIN_VERSION_base(4,11,0)
+import Prelude (Semigroup)
+#endif
+
 import Data.Function
 import Control.Monad
 import qualified Data.List as L
@@ -74,6 +80,8 @@ data Syntax =
  | Empty
    deriving Show
 
+#if MIN_VERSION_base(4,11,0)
+
 instance Semigroup Syntax where
  Empty <> x = x
  x <> Empty = x
@@ -81,6 +89,16 @@ instance Semigroup Syntax where
 
 instance Monoid Syntax where
  mempty = Empty
+
+#else
+
+instance Monoid Syntax where
+ Empty `mappend` x = x
+ x `mappend` Empty = x
+ x `mappend` y = Append x y
+ mempty = Empty
+
+#endif
 
 -- Printer
 

--- a/Text/LaTeX/Guide/Syntax.hs
+++ b/Text/LaTeX/Guide/Syntax.hs
@@ -17,7 +17,7 @@ import Data.Text
 import Data.Text.IO hiding (putStr)
 import Data.Int
 import Data.Bool
-import Prelude (Eq(..),Show(..),FilePath,Enum)
+import Prelude (Eq(..),Show(..),FilePath,Enum,Semigroup)
 import Data.Function
 import Control.Monad
 import qualified Data.List as L
@@ -74,10 +74,12 @@ data Syntax =
  | Empty
    deriving Show
 
+instance Semigroup Syntax where
+ Empty <> x = x
+ x <> Empty = x
+ x <> y = Append x y
+
 instance Monoid Syntax where
- mappend Empty x = x
- mappend x Empty = x
- mappend x y = Append x y
  mempty = Empty
 
 -- Printer

